assets/src/admin/amp-theme-install.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import domReady from '@wordpress/dom-ready';
+
+/**
+ * Internal dependencies
+ */
+import ampViewTheme from './theme-install/view/theme';
+
+const ampThemeInstall = {
+
+	/**
+	 * Init function.
+	 */
+	init() {
+		this.addTab();
+		this.overrideViews();
+	},
+
+	/**
+	 * Add a new tab for AMP-compatible themes on theme install page.
+	 */
+	addTab() {
+		const filterLinks = document.querySelector( '.filter-links' );
+		if ( ! filterLinks ) {
+			return;
+		}
+
+		const listItem = document.createElement( 'li' );
+		const anchorElement = document.createElement( 'a' );
+
+		anchorElement.setAttribute( 'href', '#' );
+		anchorElement.setAttribute( 'data-sort', 'amp-compatible' );
+		anchorElement.append( __( 'AMP Compatible', 'amp' ) );
+
+		listItem.appendChild( anchorElement );
+
+		filterLinks.appendChild( listItem );
+	},
+
+	/**
+	 * Override theme view.
+	 */
+	overrideViews() {
+		wp.themes.view.Theme = ampViewTheme;
```

Comment:
Something we may need to look out for here: what if another plugin is also doing a similar thing? Will their overrides get lost? I suppose not if they are also extending `wp.themes.view.Theme` because if they do so then they would be extending our extension, or vice-versa. Still, something to look out for.